### PR TITLE
Set the `camelcase` ESLint rule to `warn`

### PR DESCRIPTION
**Pull request by jtpio**
## References

Turning it `on` might not work just yet as some parts of the code handle names in snake case (for example server messages).

`warn` can help a bit with the GitHub lint annotations on PR diffs.

## Code changes

ESLint configuration.
Removes most of the warning to get better code and more manageable eslint output.

## User-facing changes

None

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by blink1073** [comment]
CI didn't like the rule name: `Error:   1:1  error  Definition for rule '@typescript-eslint/camelcase' was not found  @typescript-eslint/camelcase`

File: .eslintrc.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ module.exports = {
     ],
     '@typescript-eslint/no-unused-vars': ['warn', { args: 'none' }],
     '@typescript-eslint/no-use-before-define': 'off',
-    '@typescript-eslint/camelcase': 'off',
+    '@typescript-eslint/camelcase': 'warn',
```
Comment (fcollonval) [inline .eslintrc.js:41]: That rule has been removed. You need to set `naming-convention`:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md

Comment (jtpio) [reply .eslintrc.js:41]: Not sure we can have both `error` and `warn` for the same `naming-convention` rule?

File: packages/documentsearch/src/searchoverlay.tsx [hunk 2]
```diff
@@ -702,7 +706,7 @@ namespace Private {
     queryString: string,
     caseSensitive: boolean,
     regex: boolean
-  ) {
+  ): RegExp {
```
Comment (krassowski) [inline packages/documentsearch/src/searchoverlay.tsx:709]: The regular expression triggering CodeQL below was discussed in: https://github.com/jupyterlab/jupyterlab/issues/7881#issuecomment-892196340

Comment (fcollonval) [reply packages/documentsearch/src/searchoverlay.tsx:709]: Thanks for the cross reference

**Pull request merged**